### PR TITLE
Add missing methods for Database JSON RPC APIs

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Add missing methods for Database JSON RPC APIs (see https://github.com/ainblockchain/ain-blockchain/blob/master/JSON_RPC_API.md) 

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/1252

### CODE REVIEW

File: ain/db/ref.py [hunk 1]
```diff
@@ -413,6 +414,48 @@ async def matchOwner(self, params: MatchInput = None) -> Any:
             ref = Reference.extendPath(ref, getattr(params, "ref", None))
         return await self._ain.provider.send("ain_matchOwner", {"ref": ref})
 
+    async def getStateProof(self, params: StateInfoInput = None) -> Any:
+        """Fetches the state proof of a global blockchain state path.
+
+        Args:
+            params (StateProofInput): The state info input object.
```
Comment (jiyoung-an) [inline ain/db/ref.py:421]: it is different from the actual parameter type.

File: ain/db/ref.py [hunk 2]
```diff
@@ -413,6 +414,48 @@ async def matchOwner(self, params: MatchInput = None) -> Any:
...
+        ref = self._path
+        if params is not None and hasattr(params, "ref"):
+            ref = Reference.extendPath(ref, getattr(params, "ref", None))
+        return await self._ain.provider.send("ain_getStateProof", {"ref": ref})
+
+    async def getProofHash(self, params: StateInfoInput = None) -> Any:
+        """Fetches the proof hash of a global blockchain state path.
+
+        Args:
+            params (StateProofInput): The state info input object.
```
Comment (jiyoung-an) [inline ain/db/ref.py:435]: ditto

File: ain/db/ref.py [hunk 3]
```diff
@@ -413,6 +414,48 @@ async def matchOwner(self, params: MatchInput = None) -> Any:
...
+        ref = self._path
+        if params is not None and hasattr(params, "ref"):
+            ref = Reference.extendPath(ref, getattr(params, "ref", None))
+        return await self._ain.provider.send("ain_getProofHash", {"ref": ref})
+
+    async def getStateInfo(self, params: StateInfoInput = None) -> Any:
+        """Fetches the state information of a global blockchain state path.
+
+        Args:
+            params (StateProofInput): The state info input object.
```
Comment (jiyoung-an) [inline ain/db/ref.py:449]: ditto

**Review by jiyoung-an — APPROVED** [review]
LGTM with some comments! thanks!

File: ain/db/ref.py (hunk 1 above)
Comment (platfowner) [reply ain/db/ref.py]: Just fixed. Thanks!

File: ain/db/ref.py (hunk 3 above)
Comment (platfowner) [reply ain/db/ref.py]: done, thanks!

**Comment by platfowner** [comment]
Thanks for the review! @jiyoung-an Now merging..

**Pull request merged**